pkg/whitesource/scanUA.go

```diff
@@ -1,93 +1,221 @@
 
 import (
+	"encoding/json"
 	"fmt"
-	"os"
+	"path/filepath"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
 )
 
+const jvmTarGz = "jvm.tar.gz"
+const jvmDir = "./jvm"
+
 // ExecuteUAScan executes a scan with the Whitesource Unified Agent.
 func (s *Scan) ExecuteUAScan(config *ScanOptions, utils Utils) error {
+	if config.BuildTool != "mta" {
+		return s.ExecuteUAScanInPath(config, utils, ".")
+	}
+
+	log.Entry().Infof("Executing WhiteSource UA scan for MTA project")
+	pomExists, _ := utils.FileExists("pom.xml")
+	if pomExists {
+		mavenConfig := *config
+		mavenConfig.BuildTool = "maven"
+		if err := s.ExecuteUAScanInPath(&mavenConfig, utils, "."); err != nil {
+			return errors.Wrap(err, "failed to run scan for maven modules of mta")
+		}
+	} else {
+		// ToDo: only warning message?
+		//log.Entry().Warning("MTA project does not contain a pom.xml in the root. Scan results might be incomplete")
+		return fmt.Errorf("mta project does not contain an aggregator pom.xml in the root - this is mandatory")
+	}
+
+	packageJSONFiles, err := utils.FindPackageJSONFiles(config)
+	if err != nil {
+		return errors.Wrap(err, "failed to find package.json files")
+	}
+	if len(packageJSONFiles) > 0 {
+		npmConfig := *config
+		npmConfig.BuildTool = "npm"
+		for _, packageJSONFile := range packageJSONFiles {
+			// we only need the path here
+			modulePath, _ := filepath.Split(packageJSONFile)
+			projectName, err := getProjectNameFromPackageJSON(packageJSONFile, utils)
+			if err != nil {
+				return errors.Wrapf(err, "failed retrieve project name")
+			}
+			npmConfig.ProjectName = projectName
+			// ToDo: likely needs to be refactored, AggregateProjectName should only be available if we want to force aggregation?
+			s.AggregateProjectName = projectName
+			if err := s.ExecuteUAScanInPath(&npmConfig, utils, modulePath); err != nil {
+				return errors.Wrapf(err, "failed to run scan for npm module %v", modulePath)
+			}
+		}
+	}
+
+	_ = removeJre(filepath.Join(jvmDir, "bin", "java"), utils)
+
+	return nil
+}
+
+// ExecuteUAScanInPath executes a scan with the Whitesource Unified Agent in a dedicated scanPath.
+func (s *Scan) ExecuteUAScanInPath(config *ScanOptions, utils Utils, scanPath string) error {
 	// Download the unified agent jar file if one does not exist
-	if err := downloadAgent(config, utils); err != nil {
+	err := downloadAgent(config, utils)
+	if err != nil {
 		return err
 	}
 
-	// Auto generate a config file based on the working directory's contents.
-	// TODO/NOTE: Currently this scans the UA jar file as a dependency since it is downloaded beforehand
-	if err := autoGenerateWhitesourceConfig(config, utils); err != nil {
+	// Download JRE in case none is available
+	javaPath, err := downloadJre(config, utils)
+	if err != nil {
 		return err
 	}
 
+	// ToDo: Check if Download of Docker/container image should be done here instead of in cmd/whitesourceExecuteScan.go
+
+	// ToDo: check if this is required
```

Comment:
It is, if you want to use `scan.ScannedProjects` to know about actually scanned projects.